### DISCUSSION TITLE
0xc000007b error after building with OpenSSL

### DISCUSSION THREAD

**Discussion by sabadjev** [openssl/openssl#25232] [Q&A]
I am using Visual Studio 2019 to develop an x64 app on Windows and trying to build with OpenSSL 3.0.9. The build succeeds without error. However, whenever I run the code, I get a Windows error: The application was unable to start correctly (0xc000007b). This error code corresponds to missing or corrupted DLLs, or DLLs with the wrong bitness. I made sure to install with target VC-WIN64A and have checked that all the dlls and libs that I copied into my project are 64-bit (I copied the contents of C:\Program Files\OpenSSL into my project after install). I have scoured the internet looking for solutions, but nothing works. Is there any insight into why this might not be working?

**Comment by nhorman** [comment]
what application are you attempting to run, precisely, and are the dll's in the same directory as the application?

**Reply by sabadjev** [reply]
I am attempting to run our own application that I am helping develop. The dll's are in the same directory as the application.

**Reply by nhorman** [reply]
I would run dumpbin on your application to see what its DT_NEEDED entries look like, and if they resolve to what you expect

**Reply by sabadjev** [reply]
It turned out to be a 32-bit DLL that I didn't realize was an issue because it was for a different VS project. But after replacing it with a 64-bit DLL, my application launched successfully.
